src/Components/Main/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,120 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import cn from 'classnames';
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import { TodosContext } from '../../../Context/TodosContext';
+import Todos from '../../../Types/Todos';
+
+type Props = {
+  todo: Todos
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const {
+    handleCompleted,
+    handleUpdateTodo,
+    handleDeleteTodo,
+  } = useContext(TodosContext);
+
+  const [isEdit, setIsEdit] = useState(false);
+  const [editingText, setEditingText] = useState(todo.title);
+
+  const inputFocus = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputFocus.current) {
+      inputFocus.current.focus();
+    }
+  }, [isEdit]);
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter' && editingText) {
```

Comment:
enter is a magic word
Consider will be better if you create a enum for magic words and use it everywhere